blaze-client/src/main/scala/org/http4s/client/blaze/PoolManager.scala

```diff
@@ -87,19 +94,28 @@ private final class PoolManager(builder: ConnectionBuilder,
             case Some(Waiting(_, callback)) =>
               logger.debug(s"Fulfilling waiting connection request: ${stats}")
               callback(stage.right)
-            case None =>
+            case None if waitQueue.isEmpty =>
               logger.debug(s"Returning idle connection to pool: ${stats}")
               idleQueue.enqueue(stage)
+
+            case None => // this connection didn't match any pending request, kill it and start a new one for a queued request
```

Comment:
I'm not sure I agree with this one.  There is value in keeping it on hand in case another connection comes in before this one expires.  (Some pools have separate `maxOpen` and `maxIdle` as a compromise between the status quo and this, to keep a few on hand without "too many".)